app/gui/src/project-view/components/ComponentDocumentation.vue

```diff
@@ -19,20 +19,19 @@ function docsForSelection() {
   return Ok(suggestionId)
 }
 
-const displayedId = computed(() =>
-  props.displayedSuggestionId != null ? Ok(props.displayedSuggestionId) : docsForSelection(),
-)
+const docs = computed(() => docsForSelection())
+// When the selection changes, we cancel the displayed suggestion override that can be in place.
+watch(docs, (_) => (displayedId.value = null))
+
+const displayedId = computed<SuggestionId | null>({
+  get: () => overrideDisplayed.value ?? unwrapOr(docs.value, null),
+  set: (value) => (overrideDisplayed.value = value),
+})
 </script>
 
 <template>
-  <DocumentationPanel
-    v-if="displayedId?.ok"
-    :selectedEntry="displayedId.value"
-    @update:selectedEntry="emit('update:displayedSuggestionId', $event)"
-  />
-  <div v-else-if="displayedId?.ok === false" class="help-placeholder">
-    {{ displayedId.error.payload }}.
-  </div>
+  <DocumentationPanel v-if="displayedId" v-model="displayedId" />
+  <div v-else-if="!docs.ok" class="help-placeholder">{{ docs.error.payload }}.</div>
```

Comment:
It's not obvious that the `if/else-if` conditions are correct, but they are: If the value is `null`, then we would have tried to display `docs`, therefore it is correct to display the error for `docs`. Relying on the implementation details of `displayedId` here is fragile.

Reply:
True, but I don’t see a clear way to express that. I added an additional condition `!displayedId` here that can probably help a little.